tests/lib/netapplier_test.py

```diff
@@ -92,6 +93,63 @@ def test_iface_admin_state_change(netinfo_nm_mock, netapplier_nm_mock):
     )
 
 
+def test_set_sorting_multiple_ip(netinfo_nm_mock, netapplier_nm_mock):
+    config = {
+        'interfaces': [
+            {
+                'name': 'foo',
+                'type': 'unknown',
+                'state': 'up',
+                'ipv4': {
+                    'address': [
+                        {
+                            'ip': '192.168.122.10',
+                            'prefix-length': 24
+                        },
+                        {
+                            'ip': '192.168.121.10',
+                            'prefix-length': 24
+                        },
+                    ],
+                    'enabled': True
+                },
+                'ipv6': {
+                    'address': [
+                        {
+                            'ip': '2001::2',
+                            'prefix-length': 64
+                        },
+                        {
+                            'ip': '2001::1',
+                            'prefix-length': 64
+                        }
+                    ],
+                    'enabled': True
+                },
+                'mtu': 1500
+            }
+        ]
+    }
+    libnm_returned_config = copy.deepcopy(config)
+    for family in ('ipv4', 'ipv6'):
+        # In real world, libnm returned will sorted address in this case.
```

Comment:
I do not think I understand (you need to massage the English a bit).
libnm will not sort its report about the addresses (at least, not per what I understand from the discussions with thaller on this)

Reply:
I don't know why thaller insist on that. On Fedora, the output of libnm is sorted. Example(I also included this example in code comment):

```bash
[fge@fed ~]$ nmstatectl show n0 |grep 192
                        "ip": "192.168.1.1",
                        "ip": "192.168.2.1",
[fge@fed ~]$ nmcli --get-values ipv4.addresses c show dummy-n0
192.168.2.1/24, 192.168.1.1/24
```

Reply:
Based on your example here it sorted it in reverse order.
thaller mentioned it is adding it in the reverse order of how it is specified for adding.

But I think the main point is commented on line 145: If you sort it at the report step, the production code sorting is not checked, if it is broken (or get broken in the future), this test will not detect it.

As the netapplier module is so big, I am not even sure it does reach the sorting part with the proper values. This test is supposed to fail when you remove the verification ipv6 addresses sorting portion.. does it?

Reply:
This test is just simulate the failure we got when not sorting before comparing:
 * User input: `[192.168.2.1/24, 192.168.1.1/24]`
 * `netinfo.show()` got `[192.168.1.1/24, 192.168.2.1/24]`.

I don't get it why it does not detect the prolem.